### DISCUSSION TITLE
ring.cavpts: creation

### DISCUSSION THREAD

**Discussion by lnadolski** [atcollab/at#740] [Python AT]
Dear All,

Could you please tell me where ring.cavpts is defined.
Is there a special function to create it after loading a given lattice?
Shall we do it manually?

Best regards,

Laurent.

**Comment by lcarver** [comment]
Hello,
I normally just define it myself. I am not aware of any in-built function.

`cavpts = at.get_value_refpts(ring, at.RFCavity)
`
Cheers,
Lee

EDIT: This was wrong. Correct one is below.

**Reply by lnadolski** [reply]
get_value_refpts() missing 1 required positional argument: 'attrname'

**Reply by lnadolski** [reply]
In [124]: at.__version__
Out[124]: '0.5.0'

**Reply by lcarver** [reply]
Sorry this was my mistake.

`cavpts = at.get_uint32_index(ring, at.RFCavity)`

**Reply by lnadolski** [reply]
cavpts = ring.get_refpts(at.RFCavity) works as well

**Comment by lfarv** [comment]
Hi all,

Right, this attribute is really optional, it's not **necessary** to specify it. It's useful if you want to restrict the default cavity operations to a set of cavities. By default, AT takes as "main cavities" the cavities with the lowest frequency. You can set manually ring.cavpts if this assumption is not true. It may also save some time by avoiding the search for the lowest frequency. But then you are responsible for the cavpts attribute. Otherwise, forget it.

See [here](https://atcollab.github.io/at/p/howto/CavityControl.html) for explanations.
